### DISCUSSION TITLE
Is the jspsych-multi-stim-multi-response plugin still available?

### DISCUSSION THREAD

**Discussion by alaketa** [jspsych/jsPsych#2621] [Q&A]
Hi,

I was just wondering whether the **jspsych-multi-stim-multi-response plugin** is still available?
If yes, where can I download it? And if no, is there an alternative plugin for presenting a sequence of/multiple stimuli for different amounts of time within a single trial?

I am new to jsPsych so I apologize for the potential simplicity/obviousness of the question! :)

Thank you!

**Comment by jodeleeuw** [comment]
Hi @alaketa,

We stopped including that plugin with current versions of jsPsych. The reason was that usually the behavior that could be generated by it could also be generated by combining several trials together. This approach is easier for us to maintain and generally more flexible.

If you need help thinking about how to implement the trial that you want to include feel free to follow up here and share details about what you are needing to do.

**Reply by alaketa** [reply]
Dear @jodeleeuw,

Thank you so much for the super quick reply!

I am designing the AX-CPT task (with distractors), and I was thinking whether there was a better way of writing this down. 

Currently, this is what I have for an AX trial (A cue - distractor letter 1 - distractor letter 2 - distractor letter 3 - X probe) (see attached).

```js


  const jsPsych = initJsPsych();

  function distractor_generateRandomLetter() {
                const stimuli = "BCDEFGHIJLMNOPQRSTUVWZ"
                const letter = stimuli[Math.floor(Math.random() * stimuli.length)]
                const final_letter = letter.fontcolor("white")
                return final_letter.fontsize(80)
     };
  

    //Z key is "Yes", M key is "No"
    var AX_trials_1 = {
    timeline: [
        {
            type: jsPsychHtmlKeyboardResponse,
            stimulus: '<p style="font-size:50px; color:red;">A</p>',
            choices: "NO_KEYS",
            trial_duration: 1200
        },
        {
            type: jsPsychHtmlKeyboardResponse,
            stimulus: distractor_generateRandomLetter,
            choices: ['z','m'],
            trial_duration: 1200,
            response_ends_trial: false
        },
        {
            type: jsPsychHtmlKeyboardResponse,
            stimulus: distractor_generateRandomLetter,
            choices: ['z','m'],
            trial_duration: 1200,
            response_ends_trial: false
        },
        {
            type: jsPsychHtmlKeyboardResponse,
            stimulus: distractor_generateRandomLetter,
            choices: ['z','m'],
            trial_duration: 1200,
            response_ends_trial: false
        },
        {
            type: jsPsychHtmlKeyboardResponse,
            stimulus: '<p style="font-size:50px; color:red;">X</p>',
            choices: ['z','m'],
            data: {
              trial_type: 'AX'
              },
            on_finish: function(data){
            if(jsPsych.pluginAPI.compareKeys(data.response, "z")){
             data.correct = true;
            } else {
             data.correct = false; 
            }
        },
            trial_duration: 1200,
            response_ends_trial: false
        }
    ]
};

```

Again, I am **very** new to jsPsych and JavaScript in general, so I apologize in advance for the code! :)

Best,
Aleksandra

**Reply by jodeleeuw** [reply]
This looks pretty reasonable to me! I suppose you could [loop over](https://www.jspsych.org/7.2/overview/timeline/#repeating-a-set-of-trials) the distractor trial if you wanted to write it only once, but there's no real need to do that.
